### PR TITLE
#160451511 Search Functionality

### PR DESCRIPTION
**Pull request by slimsolz**
#### What does this PR do?
Allows the user search for articles based on different parameters
#### Description of Task to be completed?
- Add route to search based on the article name or the author or tags

#### How should this be manually tested?
- Clone the repo with "git clone https://github.com/andela/phlox-ah.git"
- cd into phlox-ah. cd phlox-ah
- Switch to the feature branch git branch -r then run git checkout feature/160451511/search-functionality
- Run npm install
- Rename .env_sample file to .env then fill it with your locally created Postgres database details
- If you have PostgreSQL install you can create a database on your command line with createdb your_db_name
- Run npm run migrate-reset to migrate your table into your database
- Run npm run dev to start your server
- Open postman, then visit 
- To search: `POST localhost:3000/api/v1/search`

#### What are the relevant pivotal tracker stories?
#160451511

### CODE REVIEW

**Review by oladimillion — COMMENTED** [review]
LGTM

File: server/seeders/20181010134458-demo-article.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
 module.exports = {
   up: queryInterface => queryInterface.bulkInsert('Articles', [{
-    title: 'title of article',
-    body: 'body of article',
-    slug: 'title-of-article',
-    description: 'description of article',
+    title: 'title1',
```
Comment (slimsolz) [inline server/seeders/20181010134458-demo-article.js:3]: I changed this so I could search for similar articles of different authors in my test

**Review by s-akinrele — CHANGES_REQUESTED** [review]
As discussed

File: server/helpers/article.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-
+/* eslint-disable no-restricted-globals */
```
Comment (slimsolz) [inline server/helpers/article.js:1]: Add this because eslint flagged isNaN

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**